lib/active_model/serializer/reflection.rb

```diff
@@ -60,7 +60,12 @@ def value(serializer)
         @scope = serializer.scope
 
         if block
-          instance_eval(&block)
+          block_value = instance_eval(&block)
+          if block_value == :nil
```

Comment:
This assumes that a relationship value MUST never be a `Symbol`. I guess this is a safe choice but let me know if you disagree.